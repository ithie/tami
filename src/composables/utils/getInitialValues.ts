import { ref, type Ref } from 'vue'

export default <T = Ref<number>>(
  [energy, happieness, hunger]: [number, number, number],
  noRef = false
): { energy: T; happieness: T; hunger: T } => ({
  energy: (noRef ? energy : ref(energy)) as T,
  happieness: (noRef ? happieness : ref(happieness)) as T,
  hunger: (noRef ? hunger : ref(hunger)) as T
})
