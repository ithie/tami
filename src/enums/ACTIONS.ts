export enum ACTIONS {
  FEED = 'FEED',
  PLAY = 'PLAY',
  SLEEP = 'SLEEP',
  NOTHING = 'NOTHING'
}
