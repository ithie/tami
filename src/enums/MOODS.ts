export enum MOODS {
  NEUTRAL = 'NEUTRAL',
  HUNGRY = 'HUNGRY',
  VERY_HUNGRY = 'VERY_HUNGRY',
  HAPPY = 'HAPPY',
  SUPER_HAPPY = 'SUPER_HAPPY',
  TIRED = 'TIRED'
}
