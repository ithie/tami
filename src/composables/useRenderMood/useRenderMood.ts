import { computed, ref, watchEffect, type Ref } from 'vue'
import { MOODS } from '../../enums/MOODS'

const EXPRESSIONS = {
  [MOODS.NEUTRAL]: '(o o)',
  [MOODS.HUNGRY]: '(- - )',
  [MOODS.VERY_HUNGRY]: '(x x)',
  [MOODS.HAPPY]: '(* *)',
  [MOODS.SUPER_HAPPY]: '(^ ^)',
  [MOODS.TIRED]: '(u u)'
}

export default ({
  hunger,
  happieness,
  energy
}: {
  hunger: Ref<number>
  happieness: Ref<number>
  energy: Ref<number>
}) => {
  const mood = ref(MOODS.NEUTRAL)

  watchEffect(() => {
    if (energy.value === 0) {
      mood.value = MOODS.TIRED
    } else if (hunger.value >= 9) {
      mood.value = MOODS.VERY_HUNGRY
    } else if (hunger.value <= 3) {
      mood.value = MOODS.TIRED
    } else if (hunger.value >= 7) {
      mood.value = MOODS.HUNGRY
    } else if (happieness.value >= 9) {
      mood.value = MOODS.SUPER_HAPPY
    } else if (happieness.value >= 7) {
      mood.value = MOODS.HAPPY
    } else {
      mood.value = MOODS.NEUTRAL
    }
  })

  return computed(() => EXPRESSIONS[mood.value])
}
