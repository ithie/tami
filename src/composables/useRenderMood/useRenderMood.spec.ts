import { MOODS } from '../../enums/MOODS'
import { describe, expect, test } from 'vitest'
import useRenderMood from './useRenderMood'
import { type Ref } from 'vue'
import getInitialValues from '../utils/getInitialValues'

const assertions: Array<
  [MOODS, { hunger: Ref<number>; happieness: Ref<number>; energy: Ref<number> }]
> = [
  [MOODS.TIRED, getInitialValues([0, 5, 5])],
  [MOODS.VERY_HUNGRY, getInitialValues([5, 9, 5])],
  [MOODS.TIRED, getInitialValues([5, 5, 3])],
  [MOODS.HUNGRY, getInitialValues([5, 5, 7])],
  [MOODS.SUPER_HAPPY, getInitialValues([5, 9, 5])],
  [MOODS.HAPPY, getInitialValues([5, 7, 5])],
  [MOODS.HAPPY, getInitialValues([5, 5, 5])]
]
describe('composables/useRenderMood', () => {
  assertions.forEach(([mood, given]) =>
    test(`should render mood: ${mood}`, () => {
      expect(useRenderMood(given).value).toMatchSnapshot()
    })
  )
})
