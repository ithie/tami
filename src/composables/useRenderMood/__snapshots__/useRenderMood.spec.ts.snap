// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`composables/useRenderMood > should render mood: HAPPY 1`] = `"(* *)"`;

exports[`composables/useRenderMood > should render mood: HAPPY 2`] = `"(o o)"`;

exports[`composables/useRenderMood > should render mood: HUNGRY 1`] = `"(- - )"`;

exports[`composables/useRenderMood > should render mood: SUPER_HAPPY 1`] = `"(^ ^)"`;

exports[`composables/useRenderMood > should render mood: TIRED 1`] = `"(u u)"`;

exports[`composables/useRenderMood > should render mood: TIRED 2`] = `"(u u)"`;

exports[`composables/useRenderMood > should render mood: VERY_HUNGRY 1`] = `"(^ ^)"`;
